nntrainer/graph/network_graph.cpp

```diff
@@ -712,6 +713,21 @@ NetworkGraph::finalizeContext(const std::shared_ptr<LayerNode> &lnode,
   const std::vector<Var_Grad *> &outputs =
     tensor_manager->requestOutputs(gnode, init_context.getOutputDimensions());
 
+  /**
+   * @note cache the labels and input var_grads to be able to fill them when
+   * running the graph.
+   */
+  if (gnode.getInputConnections().empty())
+    input_list.insert(input_list.end(), inputs.begin(), inputs.end());
+  /** @todo check compatibility of requireLabel() and
+   * getOutputConnections().empty() */
+  if (lnode->requireLabel())
+    label_list.insert(label_list.end(), outputs.begin(), outputs.end());
+
+  /**
+   * @note must use existing properties like name/trainable of run_context to
+   * create the new run_context
```

Comment:
There is no run context before this, could you clarify the comment here?

Reply:
yes, this must be removed after rebase. I will remove it.

Reply:
done.